src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/CompilerServices/IDOBinder.vb

```diff
@@ -1278,62 +1270,81 @@ Namespace Microsoft.VisualBasic.CompilerServices
             Return If(valueExpression.Type.Equals(GetType(Object)), valueExpression, Expression.Convert(valueExpression, GetType(Object)))
         End Function
 
+        ' MRU Dictionary of invoker delegates. We keep 16 most recently used ones, rest is GC'd
+        Private Shared Invokers As New CacheDict(Of Integer, Func(Of CallSiteBinder, Object, Object(), Object))(16)
+
         Public Shared Function CreateRefCallSiteAndInvoke(
                 ByVal action As CallSiteBinder,
                 ByVal instance As Object,
                 ByVal arguments As Object()) As Object
```

Comment:
The updated implementation of `CreateRefCallSiteAndInvoke()` is from .NET Framework.